### PR TITLE
examples/{tc,xdp}*: Use network-types crate instead of aya-tool bindings

### PR DESCRIPTION
**Pull request by vadorovsky**
Network-related structures don't change across kernel versions, so using BTF to generate them is an overkill. Instead of using aya-tool, we decided to publish a crate called network-types, which provides definitions of network headers (L2, L3 and L4) which are suitable to use as compatible with ones provided by the kernel.

Signed-off-by: Michal Rostecki <vadorovsky@gmail.com>

### CODE REVIEW

File: docs/book/start/hello-xdp.md
Comment (shaneutt) [inline docs/book/start/hello-xdp.md:5]:
```suggestion
    Full code for the example in this chapter is available [here](https://github.com/aya-rs/book/tree/main/examples/xdp-hello)
```

File: docs/book/start/logging-packets.md
Comment (shaneutt) [inline docs/book/start/logging-packets.md:9]:
```suggestion
    Full code for the example in this chapter is available [here](https://github.com/aya-rs/book/tree/main/examples/xdp-log)
```

File: docs/book/start/logging-packets.md
Comment (shaneutt) [inline docs/book/start/logging-packets.md:103]: These kinds of versions can get lost in time and cause issues for readers. Has there been any previous thoughts on doing something like templating from the `Cargo.toml` data in the relevant examples? Not at all something to hold up this PR for of course, but if there's any interest maybe I can throw together an issue for it.

**Comment by alessandrod** [comment]
I think this is good, but the network-types crate needs a bit more work before we can merge

File: examples/tc-egress/tc-egress-ebpf/src/main.rs
Comment (shaneutt) [inline examples/tc-egress/tc-egress-ebpf/src/main.rs:14]: :+1:

**Review by shaneutt — COMMENTED** [review]
Looks good, I'm really glad to see this crate available thanks for the effort :vulcan_salute:

File: docs/book/start/logging-packets.md
Comment (vadorovsky) [reply docs/book/start/logging-packets.md]: Good point. I removed the version from the sentence, we show the whole `Cargo.toml` content below anyway, so I think that should suffice.

File: docs/book/start/hello-xdp.md
Comment (vadorovsky) [reply docs/book/start/hello-xdp.md]: done

File: docs/book/start/logging-packets.md
Comment (vadorovsky) [reply docs/book/start/logging-packets.md]: done

File: docs/book/start/logging-packets.md
Comment (vadorovsky) [reply docs/book/start/logging-packets.md]: done

**Comment by vadorovsky** [comment]
Let's wait for https://github.com/vadorovsky/network-types/pull/3 to get merged before this one

**Comment by vadorovsky** [comment]
I updated the PR to use the newest network-types.

**Review by dave-tucker — APPROVED** [review]
LGTM

**Review by shaneutt — APPROVED** [review]

**Pull request merged**
